src/Neo/SmartContract/Native/ContractMethodAttribute.cs

```diff
@@ -9,15 +9,18 @@
 // modifications are permitted.
 
 using System;
+using System.Diagnostics;
 
 namespace Neo.SmartContract.Native
 {
+    [DebuggerDisplay("{Name}")]
     [AttributeUsage(AttributeTargets.Method | AttributeTargets.Property, AllowMultiple = false)]
     internal class ContractMethodAttribute : Attribute
     {
         public string Name { get; init; }
         public CallFlags RequiredCallFlags { get; init; }
         public long CpuFee { get; init; }
         public long StorageFee { get; init; }
+        public Hardfork? ActiveIn { get; init; } = null;
```

Comment:
Looks good, @shargon. But what if we need to update the behavior of a contract method? And we need to keep its name and parameters. Then, we would need different versions of the method.

Reply:
Always we can use `IsHArdForkEnabled` inside the logic

Reply:
I see, that will be the case.

Reply:
I think we should improve that, @shargon.
Maybe it is possible to retrieve a specific version from storage and run it.

The neo-vm will require something similar.

Reply:
> I think we should improve that, @shargon. Maybe it is possible to retrieve a specific version from storage and run it.
> 
> The neo-vm will require something similar.

Native contracts logic are not stored

Reply:
I know, but we could store it and the abi for syncing purpose and re-run.
And a similar strategy could be done for the neo-vm. I think that if we go this way we will be fully safe and simple in terms of updates. No complex logics.